todo-app/src/App.js

```diff
@@ -1,21 +1,107 @@
-import React, { Component } from 'react';
+import React from 'react';
 import logo from './logo.svg';
 import './App.css';
+import List from './list'
 
-class App extends Component {
-  render() {
-    return (
-      <div className="App">
-        <div className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h2>Welcome to React</h2>
+  class App extends React.Component {
+    constructor (props){
+        super(props);
+        this.state = {
+          todos: [
+               {
+                description: "Read more about React",
+                done:        true
+               } ,
+               {
+              description: "Do the homework",
+               done:        false
+              },
+            ],
+          newToDo:{
+            description: "",
+            done:        false
+           }
+        }
+
+    }
+
+    handleSubmit(event){
+      event.preventDefault()
+      const {newToDo} = this.state  
+      const todos = [...this.state.todos, newToDo]
+      this.setState({todos})
+      document.getElementById("addItem").reset();
+      }
+
+    handleChange(event){
+      this.setState({
+        newToDo: {
+          description: event.target.value,
+          done: false
+        }
+      })
+    }
+
+    renderAddItem (){
+      return(
+        <div>
+          <form id="addItem" onSubmit={this.handleSubmit.bind(this)}>
+            <input type="submit" value="Add" />
+            <input type="text"  onChange={this.handleChange.bind(this)}/>
+          </form>
         </div>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
-      </div>
-    );
+      )
+    }
+
+    onDelete= (index) => {
+      const todosList = this.state.todos
+      todosList.splice(index, 1)
+      this.setState({todosList})
+    }
+
+
+
+    handleCheckBoxChange = (index) => {
+      const todosList = this.state.todos
+      todosList[index].done = !todosList[index].done;
+      this.setState({
+        todosList
+      });
+    }
+
+    save = (editToDo,index) => {
+      const todosList = this.state.todos
+      todosList[index] = editToDo
+      this.setState({
+        todosList
+        });
+    }
+
+    render() {
+      let todoList=this.state.todos
+      todoList= todoList.map(function(item,index){
+        return( <List  description={item.description}
```

Comment:
I am not sure why you used another component here `List` that just renders a `ListItem`?  If you check your react dev tools with chrome, this code renders two `List` components.

<img width="458" alt="screen shot 2017-07-21 at 21 57 14" src="https://user-images.githubusercontent.com/1057324/28480162-af20ee8a-6e5f-11e7-9c66-47ea5b7b6bc1.png">

What you probably wanted to do here is to render one `List` component and pass as a prop the `todoList`. The `List` component then would render in a map the `ListItem` components. So this code would end up being something like

```javascript
<List
  todos={todoList}
  onDelete={this.onDelete}
  handleCheckBoxChange={this.handleCheckBoxChange}
  save={this.save}
/>
```

Does this make sense? Do you think this is a better approach?